hybrid/ddc_compartment_switching/ddc_compartment_switching_nok.c

```diff
@@ -34,8 +34,7 @@ int compartment_simple_fun()
 {
 	int32_t *__capability ddc_cap = cheri_ddc_get();
 	// This function can access only 2000 bytes, i.e. `compartment_size`
-	assert(cheri_tag_get(ddc_cap) && cheri_length_get(ddc_cap) == 2000);
-	// This function cannot go over its compartment bounds
-	assert(ddc_cap[700]);
+	// So the following will go over its compartment bounds
+	assert(ddc_cap[700] == 8);
```

Comment:
What happens when you run this line? Does it SIGPROT?

Reply:
Yes.

Reply:
Then the `assert` is misleading: it's never executed, right?

Reply:
If we remove the `assert` it does not SIGPROT, though.

Reply:
> Then the `assert` is misleading: it's never executed, right?

Why would the `assert` not be executed? The access is out of bounds, and both versions try to load from it. (The old assertion is just `ddc_cap[700] != 0`.)